instana/instrumentation/asgi.py

```diff
@@ -21,11 +21,12 @@ def __init__(self, app):
 
     def _extract_custom_headers(self, span, headers):
         try:
-            for custom_header in agent.options.extra_http_headers:
-                # Headers are in the following format: b'x-header-1'
-                for header_pair in headers:
-                    if header_pair[0].decode('utf-8').lower() == custom_header.lower():
-                        span.set_tag("http.header.%s" % custom_header, header_pair[1].decode('utf-8'))
+            if agent.options.extra_http_headers is not None:
```

Comment:
This check should be inverted for an early return and should go outside of the try block.

Reply:
+1 for Ferenc's comment. You can follow the same as you did for Django's PR.